localfs_windows.go

```diff
@@ -0,0 +1,125 @@
+// +build windows
+
+package desync
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// LocalFS uses the local filesystem for tar/untar operations.
+type LocalFS struct {
+	// Base directory
+	Root string
+
+	opts LocalFSOptions
+
+	dev     uint64
+	once    sync.Once
+	entries chan walkEntry
+	sErr    error
+}
+
+// LocalFSOptions influence the behavior of the filesystem when reading from or writing too it.
+type LocalFSOptions struct {
+	// Only used when reading from the filesystem. Will only return
+	// files from the same device as the first read operation.
+	OneFileSystem bool
+
+	// When writing files, use the current owner and don't try to apply the original owner.
+	NoSameOwner bool
+
+	// Ignore the incoming permissions when writing files. Use the current default instead.
+	NoSamePermissions bool
+
+	// Reads all timestamps as zero. Used in tar operations to avoid unneccessary changes.
+	NoTime bool
+}
+
+var _ FilesystemWriter = &LocalFS{}
+var _ FilesystemReader = &LocalFS{}
```

Comment:
I'm wondering if we could move the code that's common between OS' into a file that's included in all builds to avoid the duplication. Basically have three files:
- localfs.go - common to all builds, would contain things like the structures
- localfs_windows.go - Windows specific file creation and reading
- localfs_other.go - Non-windows via build tags in the header